gapic-generator-cloud/lib/gapic/generators/cloud_generator.rb

```diff
@@ -42,17 +43,15 @@ def initialize api
       # @return [Array<
       #   Google::Protobuf::Compiler::CodeGeneratorResponse::File>]
       #   The files that were generated for the API.
-      def generate
+      def generate gem_presenter: nil
         gem_config = @api.configuration[:gem] ||= {}
         return generate_wrapper if gem_config[:version_dependencies]
 
-        orig_files = super
-
-        cloud_files = []
-
-        gem = Gapic::Presenters.gem_presenter @api
+        gem = gem_presenter || Gapic::Presenters.cloud_gem_presenter(@api)
```

Comment:
What is the reason for this factory method instead of simply calling  `Gapic::Presenters::CloudGemPresenter.new(@api)`?

Reply:
Honestly, I didn't have a reason beyond following the existing pattern—and I don't know that there's a compelling reason for that pattern in the first place. We can simplify it at some point.